downstairs/src/region.rs

```diff
@@ -1,1565 +1,81 @@
-                .map(|(block, hash)| (*block, hash.unwrap())),
-            &tx,
-        )?;
-
-        cdt::extent__flush__sqlite__insert__done!(|| {
-            (job_id.get(), self.number, n_dirty_blocks)
-        });
-
-        inner.set_flush_number(new_flush, new_gen)?;
-        tx.commit()?;
-        inner.dirty_blocks.clear();
-
-        // Finally, reset the file's seek offset to 0
-        inner.file.seek(SeekFrom::Start(0))?;
-
-        cdt::extent__flush__done!(|| {
-            (job_id.get(), self.number, n_dirty_blocks)
-        });
-
-        Ok(())
-    }
-
-    /// Rehash the entire file. Remove any stored hash/encryption contexts
-    /// that do not correlate to data currently stored on disk. This is
-    /// primarily when opening an extent after recovering from a crash, since
-    /// irrelevant hashes will normally be cleared out during a flush().
-    ///
-    /// By default this function will only do work if the extent is marked
-    /// dirty. Set `force_override_dirty` to `true` to override this behavior.
-    /// This override should generally not be necessary, as the dirty flag
-    /// is set before any contexts are written.
-    #[instrument]
-    async fn fully_rehash_and_clean_all_stale_contexts(
-        &self,
-        force_override_dirty: bool,
-    ) -> Result<(), CrucibleError> {
-        let mut inner = self.inner.lock().await;
-
-        if !force_override_dirty && !inner.dirty()? {
-            return Ok(());
-        }
+use super::*;
+use crate::extent::{
+    copy_dir, extent_dir, extent_file_name, move_replacement_extent,
+    replace_dir, sync_path, Extent, ExtentMeta, ExtentState, ExtentType,
+};
 
-        // Just in case, let's be very sure that the file on disk is what it should be
-        if let Err(e) = inner.file.sync_all() {
-            crucible_bail!(
-                IoError,
-                "extent {}: fsync 1 failure during full rehash: {:?}",
-                self.number,
-                e
-            );
-        }
+/**
+ * Validate a list of sorted repair files.
+ * There are either two or four files we expect to find, any more or less
+ * and we have a bad list.  No duplicates.
+ */
+pub fn validate_repair_files(eid: usize, files: &[String]) -> bool {
+    let eid = eid as u32;
 
-        inner.file.seek(SeekFrom::Start(0))?;
+    let some = vec![
+        extent_file_name(eid, ExtentType::Data),
+        extent_file_name(eid, ExtentType::Db),
+    ];
 
-        // Buffer the file so we dont spend all day waiting on syscalls
-        let mut inner_file_buffered =
-            BufReader::with_capacity(64 * 1024, &inner.file);
+    let mut all = some.clone();
+    all.extend(vec![
+        extent_file_name(eid, ExtentType::DbShm),
+        extent_file_name(eid, ExtentType::DbWal),
+    ]);
 
-        // This gets filled one block at a time for hashing
-        let mut block = vec![0; self.block_size as usize];
+    // Either we have some or all.
+    files == some || files == all
```

Comment:
This is identical to the [current implementation](https://github.com/oxidecomputer/crucible/blob/20273bcca1fd5834ebc3e67dfa7020f0e99ad681/downstairs/src/region.rs#L533-L554), but I can change it to a hashset comparison if you want!

(I'm inclined to leave it, because once we switch to the new SQLite-free backend, there's only one file and this becomes trivial)